### PR TITLE
Add Oauth2 reference client + example

### PR DESCRIPTION
**Pull request by mboylevt**

### CODE REVIEW

**Comment by brettlangdon** [comment]
Please run your files through [gofmt](https://golang.org/cmd/gofmt/) :)

Should just need to run `gofmt` in the source folder.

**Comment by mboylevt** [comment]
@brettlangdon hah, seems I forgot to do that.  On it ;)

File: example/oauth2_example.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package main
+import "fmt"
+import "../shapeways_oauth2/"
```
Comment (brettlangdon) [inline example/oauth2_example.go]: Should have this as `github.com/Shapeways/go-shapeways/shapeways_oauth2`

File: shapeways_oauth2/shapeways_oauth2.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  return client
+}
+
+// Helper function to perform request and handle response
+func (client *Oauth2Client) DoHttpRequest(req *http.Request) (*http.Response, error) {
+  http_client := &http.Client{}
+  resp, err := http_client.Do(req)
+  check(err)
+  if resp.StatusCode != 200 {
+    fmt.Println("Request returned status " + resp.Status)
```
Comment (brettlangdon) [inline shapeways_oauth2/shapeways_oauth2.go]: Probably want to remove this

**Comment by mboylevt** [comment]
...and it seems i've boned the wrong button.  hang on, fixing.

**Comment by brettlangdon** [comment]
lol

**Pull request merged**